### ISSUE TITLE
Link error while building on Windows

### ISSUE THREAD

**Issue by aemino** [twistedfall/opencv-rust#60]
Hello there, I'm encountering an error while trying to build `opencv = {version = "0.24.5", default-features = false, features = ["opencv-41", "buildtime-bindgen"]}` on Windows with the suggested environment variables.

Here's that error:

```
error: linking with `link.exe` failed: exit code: 1120
  |
  = note: "C:\\Program Files (x86)\\Microsoft Visual Studio\\2017\\BuildTools\\VC\\Tools\\MSVC\\14.13.26128\\bin\\HostX64\\x64\\link.exe" "/NOLOGO" "/NXCOMPAT" "/LIBPATH:C:\\Users\\aemino\\.rustup\\toolchains\\nightly-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib" "D:\\Git\\toonfish\\target\\debug\\deps\\toonfish.14fbwp9wfczr4lrm.rcgu.o" "D:\\Git\\toonfish\\target\\debug\\deps\\toonfish.1oeylic2ccszagg0.rcgu.o" "D:\\Git\\toonfish\\target\\debug\\deps\\toonfish.3a6y7s2wqp1nkhlo.rcgu.o" "D:\\Git\\toonfish\\target\\debug\\deps\\toonfish.3dbsph93yihankc8.rcgu.o" "D:\\Git\\toonfish\\target\\debug\\deps\\toonfish.50nc50dulfdg8e2e.rcgu.o" "D:\\Git\\toonfish\\target\\debug\\deps\\toonfish.5botmstpj0b4hxk4.rcgu.o" "D:\\Git\\toonfish\\target\\debug\\deps\\toonfish.966r1upqsysg8sm.rcgu.o" "/OUT:D:\\Git\\toonfish\\target\\debug\\deps\\toonfish.exe" "D:\\Git\\toonfish\\target\\debug\\deps\\toonfish.4dsp84ed06uti2eb.rcgu.o" "/OPT:REF,NOICF" 
"/DEBUG" "/NATVIS:C:\\Users\\aemino\\.rustup\\toolchains\\nightly-x86_64-pc-windows-msvc\\lib\\rustlib\\etc\\intrinsic.natvis" "/NATVIS:C:\\Users\\aemino\\.rustup\\toolchains\\nightly-x86_64-pc-windows-msvc\\lib\\rustlib\\etc\\liballoc.natvis" "/NATVIS:C:\\Users\\aemino\\.rustup\\toolchains\\nightly-x86_64-pc-windows-msvc\\lib\\rustlib\\etc\\libcore.natvis" "/LIBPATH:D:\\Git\\toonfish\\target\\debug\\deps" "/LIBPATH:D:\\Git\\toonfish\\target\\debug\\build\\opencv-51cfe1331c65ee8c\\out" "/LIBPATH:D:\\OpenCV\\build\\x64\\vc14\\lib" "/LIBPATH:D:\\Git\\toonfish\\target\\debug\\build\\opencv-51cfe1331c65ee8c\\out" "/LIBPATH:C:\\Users\\aemino\\.rustup\\toolchains\\nightly-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib" "D:\\Git\\toonfish\\target\\debug\\deps\\libscrap-6fb5b0a62edbeed3.rlib" "D:\\Git\\toonfish\\target\\debug\\deps\\libwinapi-bf08ae65de4518c6.rlib" "D:\\Git\\toonfish\\target\\debug\\deps\\libcfg_if-af76bf5447c4b33d.rlib" "D:\\Git\\toonfish\\target\\debug\\deps\\libopencv-0227b599a1a7f4a5.rlib" "D:\\Git\\toonfish\\target\\debug\\deps\\libslab-ba76652ae6434093.rlib" "D:\\Git\\toonfish\\target\\debug\\deps\\liblazy_static-78f0258cdacfac46.rlib" "D:\\Git\\toonfish\\target\\debug\\deps\\libnum-072536065608753c.rlib" "D:\\Git\\toonfish\\target\\debug\\deps\\libnum_rational-302080e511cb6aad.rlib" "D:\\Git\\toonfish\\target\\debug\\deps\\libnum_iter-776e5ab758163e5e.rlib" "D:\\Git\\toonfish\\target\\debug\\deps\\libnum_complex-7e44487630133f86.rlib" "D:\\Git\\toonfish\\target\\debug\\deps\\libnum_bigint-f7a4e9f30baad745.rlib" "D:\\Git\\toonfish\\target\\debug\\deps\\libnum_integer-83c8439276e95286.rlib" "D:\\Git\\toonfish\\target\\debug\\deps\\libnum_traits-880db1d3befbd7f4.rlib" "D:\\Git\\toonfish\\target\\debug\\deps\\liblibc-17ef0b4ff3f7910d.rlib" "D:\\Git\\toonfish\\target\\debug\\deps\\libcpp-e1df142f37e70443.rlib" "C:\\Users\\aemino\\.rustup\\toolchains\\nightly-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libstd-e23fccb0a8b0ef4d.rlib" "C:\\Users\\aemino\\.rustup\\toolchains\\nightly-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libpanic_unwind-084692af5936cf7a.rlib" "C:\\Users\\aemino\\.rustup\\toolchains\\nightly-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libhashbrown-d384eae24131de71.rlib" "C:\\Users\\aemino\\.rustup\\toolchains\\nightly-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\librustc_std_workspace_alloc-6f4dd817315b8a2e.rlib" "C:\\Users\\aemino\\.rustup\\toolchains\\nightly-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libbacktrace-6705ac3c1c6fa52a.rlib" "C:\\Users\\aemino\\.rustup\\toolchains\\nightly-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\librustc_demangle-a713aa7f7e6f8b9c.rlib" "C:\\Users\\aemino\\.rustup\\toolchains\\nightly-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libunwind-722e47fd51e21662.rlib" "C:\\Users\\aemino\\.rustup\\toolchains\\nightly-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libcfg_if-d5ead5144bca211f.rlib" "C:\\Users\\aemino\\.rustup\\toolchains\\nightly-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\liblibc-44b0c7d4e4124c1b.rlib" "C:\\Users\\aemino\\.rustup\\toolchains\\nightly-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\liballoc-349ac95fc5d39ffb.rlib" "C:\\Users\\aemino\\.rustup\\toolchains\\nightly-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\librustc_std_workspace_core-734958cc1e87db6a.rlib" "C:\\Users\\aemino\\.rustup\\toolchains\\nightly-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libcore-148ef0fa79c16b3f.rlib" "C:\\Users\\aemino\\.rustup\\toolchains\\nightly-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libcompiler_builtins-6fbdc4cdd29b327d.rlib" "dxgi.lib" "d3d11.lib" "advapi32.lib" "ws2_32.lib" "userenv.lib" "msvcrt.lib"
  = note: libopencv-0227b599a1a7f4a5.rlib(stitching.o) : error LNK2019: unresolved external symbol "struct cv::Ptr<class cv::Stitcher> __cdecl cv::createStitcher(bool)" (?createStitcher@cv@@YA?AU?$Ptr@VStitcher@cv@@@1@_N@Z) referenced in function cv_createStitcher_bool
          libopencv-0227b599a1a7f4a5.rlib(stitching.o) : error LNK2019: unresolved external symbol "struct cv::Ptr<class cv::Stitcher> __cdecl cv::createStitcherScans(bool)" (?createStitcherScans@cv@@YA?AU?$Ptr@VStitcher@cv@@@1@_N@Z) referenced in function cv_createStitcherScans_bool
          libopencv-0227b599a1a7f4a5.rlib(dnn.o) : error LNK2019: unresolved external symbol "public: __cdecl cv::dnn::dnn4_v20190621::BackendNode::BackendNode(int)" (??0BackendNode@dnn4_v2019062621@dnn@cv@@QEAA@H@Z) referenced in function cv_dnn_BackendNode_BackendNode_int
          D:\Git\toonfish\target\debug\deps\toonfish.exe : fatal error LNK1120: 3 unresolved externals
```

I noticed that @Weasy666 was also having this error despite not using the `dnn` or `stitching` modules. Using `--release` mode does only seems to get rid of the unresolved `dnn` external, not the two `stitching` ones for whatever reason.

Let me know if any other information would be useful.

**Comment by aemino** [comment]
I was able to resolve this error by removing the following lines from `stitching.hpp`:
```
/**
 * @deprecated use Stitcher::create
 */
CV_DEPRECATED Ptr<Stitcher> createStitcher(bool try_use_gpu = false);

/**
 * @deprecated use Stitcher::create
 */
CV_DEPRECATED Ptr<Stitcher> createStitcherScans(bool try_use_gpu = false);
```

It seems these definitions have been removed in OpenCV 4.1.x?

**Comment by twistedfall** [comment]
That's weird, what OpenCV version are you using and how did you install it?

**Comment by aemino** [comment]
I tried both 4.1.1 (currently using) and 4.1.2. I got my installation from the OpenCV GitHub release page; I unzipped it and then added the various paths as environment variables.

I was tipped off to remove those methods because they don't appear in the OpenCV docs from ~4.0.0 onward.

**Comment by twistedfall** [comment]
Well as far as I can see they are present in the docs for 4.1.2: https://docs.opencv.org/4.1.2/d1/d46/group__stitching.html#ga2de00132401436e7ddc89590994fd922

Will you be able to confirm that those symbols are really missing in the opencv dll? With something like http://www.dependencywalker.com/

**Comment by aemino** [comment]
[Here](https://github.com/twistedfall/opencv-rust/files/3825859/opencv_methods.txt) is a dump of methods in `build/x64/vc15/bin/opencv_world411.dll` from [Dependencies](https://https://github.com/lucasg/Dependencies) (Dependency Walker didn't work for me). It seems that the `createStitcher` methods are missing, unless I just failed to find them.

**Comment by aemino** [comment]
@twistedfall Just pinging in case you had forgotten about this issue :)

**Comment by twistedfall** [comment]
There is some work happening about building on Windows in https://github.com/twistedfall/opencv-rust/pull/69 that might help in your case. You can expect an update today or tomorrow.

**Comment by twistedfall** [comment]
These 3 linking issues should be fixed in 0.28.0
